core/src/credential/validation.rs

```diff
@@ -0,0 +1,32 @@
+use super::ContextValue;
+
+// Context must contain at least one URI
+// The first URI must be https://www.w3.org/2018/credentials/v1 (use BASE_CREDENDIAL_CONTEXT)
+pub fn credential_context_validation(
+    val: &Vec<ContextValue>,
+) -> Result<(), serde_valid::validation::Error> {
+    if val.is_empty() {
+        return Err(serde_valid::validation::Error::Custom(
+            "Context must contain at least one URI".to_string(),
+        ));
+    }
+
+    match val[0] {
+        ContextValue::String(ref s) => {
+            if s != super::BASE_CREDENTIAL_CONTEXT {
```

Comment:
This check can be part of the match statement itself:

```
ContextValue::String(ref s) if s != super::BASE_CREDENTIAL_CONTEXT => {
   return Err(...)
```

Reply:
@gshuflin The problem with this is that then I have to create another match arm to cover where the context value is a string but not the conditional isn't, which ends up being longer than if I just had this